lib/rubocop/cop/lint/nested_method_definition.rb

```diff
@@ -27,7 +27,7 @@ class NestedMethodDefinition < Cop
         PATTERN
 
         def_node_matcher :class_or_module_new_call?, <<-PATTERN
```

Comment:
This isn't an accurate name for the node matcher after my change, but I didn't like the idea of just making it `:class_or_module_or_struct_new_call?`, so I am hoping someone else has a better name.